q2_types/per_sample_sequences/_transformer.py

```diff
@@ -135,3 +140,222 @@ def _5(dirfmt: SingleLanePerSamplePairedEndFastqDirFmt) \
+                         'following sample ids were associated with more '
+                         'than one forward read record: '
+                         '%s' % ', '.join(duplicated_ids_forward))
+
+    duplicated_ids_reverse = _duplicated_ids(reverse_direction_sample_ids)
+    if len(duplicated_ids_reverse) > 0:
+        raise ValueError('Each sample id can have only one reverse read '
+                         'record in a paired-end read manifest, but the '
+                         'following sample ids were associated with more '
+                         'than one reverse read record: '
+                         '%s' % ', '.join(duplicated_ids_reverse))
+
+    if sorted(forward_direction_sample_ids) != \
+       sorted(reverse_direction_sample_ids):
+
+        forward_but_no_reverse = set(forward_direction_sample_ids) - \
+            set(reverse_direction_sample_ids)
+
+        reverse_but_no_forward = set(reverse_direction_sample_ids) - \
+            set(forward_direction_sample_ids)
+
+        if len(forward_but_no_reverse) > 0:
+            raise ValueError('Forward and reverse reads must be provided '
+                             'exactly one time each for each sample. The '
+                             'following samples had forward but not '
+                             'reverse read fastq files: %s'
+                             % ', '.join(forward_but_no_reverse))
+        else:
+            raise ValueError('Forward and reverse reads must be provided '
+                             'exactly one time each for each sample. The '
+                             'following samples had reverse but not '
+                             'forward read fastq files: %s'
+                             % ', '.join(reverse_but_no_forward))
+
+
+def _fastq_manifest_helper(fmt, fastq_copy_fn, single_end):
+    input_manifest = _parse_and_validate_manifest(fmt.open(),
+                                                  single_end=single_end)
+    if single_end:
+        result = SingleLanePerSampleSingleEndFastqDirFmt()
+    else:
+        result = SingleLanePerSamplePairedEndFastqDirFmt()
+
+    output_manifest_data = []
+    for idx, sample_id, input_fastq_fp, direction in \
+            input_manifest.itertuples():
+        output_fastq_fn = \
+            _get_fastq_fn(sample_id, idx, direction)
+        output_manifest_data.append([sample_id, output_fastq_fn, direction])
+        output_fastq_fp = str(result.path / output_fastq_fn)
+        fastq_copy_fn(input_fastq_fp, output_fastq_fp)
+
+    output_manifest = FastqManifestFormat()
+    output_manifest_df = pd.DataFrame(
+        output_manifest_data, columns=['sample-id', 'filename', 'direction'])
+    output_manifest_df.to_csv(str(output_manifest), index=False)
+    result.manifest.write_data(output_manifest, FastqManifestFormat)
+
+    metadata = YamlFormat()
+    metadata.path.write_text(yaml.dump({'phred-offset': 33}))
+    result.metadata.write_data(metadata, YamlFormat)
```

Comment:
Not really worth it (and it looks like the other transformer is doing the same thing as you), but you could define a transformer from `object -> YamlFormat` then you could drop the line just above this one with:
`result.metadata.write_data({'phred-offset': 33}, YamlFormat)`

Same could be done for `pd.DataFrame -> FastqManifestFormat`.

Basically `write_data` can invoke other transformers (just like `Artifact.import_data`)